allennlp/data/semparse/type_declarations/wikitables_type_declaration.py

```diff
@@ -65,66 +66,41 @@ def substitute_any_type(self, basic_types: Set[BasicType]) -> List[Type]:
         return substitutions
 
 
-class ConjunctionType(PlaceholderType):
-    """
-    ``ConjunctionType`` takes an entity of any type and returns a function that takes and returns the same
-    type. That is, its signature is <#1, <#1, #1>>
-    """
-    @property
-    def _signature(self) -> str:
-        return "<#1,<#1,#1>>"
-
-    @overrides
-    def resolve(self, other: Type) -> Optional[Type]:
-        """See ``PlaceholderType.resolve``"""
-        if not isinstance(other, NltkComplexType):
-            return None
-        if not isinstance(other.second, NltkComplexType):
-            return None
-        other_first = other.first.resolve(other.second.first)
-        if other_first is None:
-            return None
-        other_first = other_first.resolve(other.second.second)
-        if not other_first:
-            return None
-        other_second = other.second.resolve(ComplexType(other_first, other_first))
-        if not other_second:
-            return None
-        return ConjunctionType(other_first, other_second)
-
-    @overrides
-    def get_application_type(self, argument_type: Type) -> Type:
-        return ComplexType(argument_type, argument_type)
-
-    @overrides
-    def substitute_any_type(self, basic_types: Set[BasicType]) -> List[Type]:
-        if self.first != ANY_TYPE:
-            return [self]
-        return [ConjunctionType(basic_type, ComplexType(basic_type, basic_type)) for basic_type in basic_types]
-
-
 class ArgExtremeType(PlaceholderType):
     """
     This is the type for argmax and argmin in Sempre. The type signature is <d,<d,<#1,<<d,#1>,#1>>>>.
     Example: (argmax (number 1) (number 1) (fb:row.row.league fb:cell.usl_a_league) fb:row.row.index)
     meaning, of the subset of rows where league == usl_a_league, find the row with the maximum index.
     """
+    def __init__(self, basic_type: BasicType = ANY_TYPE, lambda_arg_type: BasicType = ANY_TYPE) -> None:
+        super().__init__(NUMBER_TYPE,
+                         ComplexType(NUMBER_TYPE,
+                                     ComplexType(basic_type,
+                                                 ComplexType(ComplexType(lambda_arg_type, basic_type),
+                                                             basic_type))))
+
     @property
     def _signature(self) -> str:
-        return "<d,<d,<#1,<<d,#1>,#1>>>>"
+        return "<n,<n,<#1,<<#2,#1>,#1>>>>"
```

Comment:
Yeah, a type hierarchy would have solved this, maybe.  It might have also complicated things while parsing...  I'd vote for worrying about this later, if/when we make more sweeping changes to nltk's logic.  The next thing we do might be just parsing code, anyway, and we may not need nltk at all for that.